tools/mmp/driver.cs

```diff
@@ -771,12 +773,35 @@ static void Pack (IList<string> unprocessed)
 			if (App.LinkMode != LinkMode.All && App.RuntimeOptions != null)
 				App.RuntimeOptions.Write (App.AppDirectory);
 
+			if (experimental_xm_aot.HasValue && experimental_xm_aot.Value && IsUnified) {
+				AOTBundle ();
+				Watch ("AOT Compile", 1);
+			}
+
 			if (!string.IsNullOrEmpty (certificate_name)) {
 				CodeSign ();
 				Watch ("Code Sign", 1);
 			}
 		}
 
+		static void AOTBundle ()
+		{
+			string monoExe = Is64Bit ? "/usr/local/bin/mono64" : "/usr/local/bin/mono";
+			var dirInfo = new DirectoryInfo (mmp_dir);
+
+			Func <FileInfo, bool> aotableFile = x => {
+				string extension = x.Extension.ToLowerInvariant ();
+				return extension == ".exe" || extension == ".dll";
+			};
+
+			string quotedFileList = string.Join (" ", dirInfo.EnumerateFiles ().Where (aotableFile).Select (x => Quote(x.ToString ())));
+			foreach (var file in dirInfo.EnumerateFiles ().Where (aotableFile))
+			{
+				if (RunCommand (monoExe, String.Format ("--aot=hybrid {0}", Quote (file.ToString ())), new string [] {"MONO_PATH", mmp_dir }) > 0)
```

Comment:
This can be parallelized to speed it up significantly.

mtouch currently uses `Parallel.ForEach` to AOT in parallel [[1]] (the default is to set MaxDegreeOfParallelism to the number of processors, but this can be overridden by passing -jX /--jobs X (like for make) to motuch).

[1]: https://github.com/xamarin/xamarin-macios/blob/c9b440ea5c5c38d1fd54b5f646eefc4adca5e47b/tools/mtouch/Application.cs#L1597